python/test/unit/language/test_core.py

```diff
@@ -5106,3 +5106,30 @@ def test_tl_range(device):
             ptx = pgm.asm['ptx']
             # check that the loop got pipelined with the right number of stages.
             assert 'cp.async.wait_group 0x6' in ptx
+
+
+@pytest.mark.parametrize("dtype, M, N, K, Axis", [(dtype, M, N, K, Axis)
+                                                  for dtype in ['float32']
+                                                  for M in [16, 32, 64]
+                                                  for N in [16, 32, 64]
+                                                  for K in [16, 32, 64]
+                                                  for Axis in [0, 1, 2]])
```

Comment:
Should I reduce the number of tested shapes? Currently, this test runs 6 minutes on my machine.